src/commands/cmd_server.cc

```diff
@@ -289,6 +289,13 @@ class CommandDisk : public Commander {
     auto s = disk_db.Type(args_[2], &type);
     if (!s.ok()) return {Status::RedisExecErr, s.ToString()};
 
+    // Redis returns the Nil string when the key does not exist,
+    // so we need to check the type here to check if the key is not exists.
+    if (type == RedisType::kRedisNone) {
+      *output = redis::NilString();
+      return Status::OK();
+    }
+
     uint64_t result = 0;
     s = disk_db.GetKeySize(args_[2], type, &result);
     if (!s.ok()) return {Status::RedisExecErr, s.ToString()};
```

Comment:
I would recommend checking the returned status after the call to the `GetKeySize` as well (or only after this call) because, in theory, something can happen between the call to `Type` and `GetKeySize` (key deletion, expiration, etc.).

Reply:
@git-hulk Thank you for quick fix :+1:

Reply:
Sounds ok, RocksDB Snapshot might helps for this, but I after `GetKeySize` would be more easily

Reply:
> I would recommend checking the returned status after the call to the `GetKeySize` as well (or only after this call) because, in theory, something can happen between the call to `Type` and `GetKeySize` (key deletion, expiration, etc.).

good suggestion.